File: src/AtCoder/abc245/d.rb
# frozen_string_literal: true

N, M = gets.chomp.split.map(&:to_i)
A = gets.chomp.split.map(&:to_i)
C = gets.chomp.split.map(&:to_i)

A.reverse!
C.reverse!
ans = []

while C.length >= A.length
  quotient = C.first / A.first
  ans << quotient
  A.each_with_index do |a, idx|
    C[idx] -= a * quotient
  end
  C.shift
end

puts ans.reverse!.join(' ')

File: src/AtCoder/abc245/d.py
N, M = map(int, input().split())
A = list(map(int, input().split()))
C = list(map(int, input().split()))

A.reverse()
C.reverse()
ans = []

while len(C) >= len(A):
    quotient = C[0] // A[0]
    ans.append(quotient)
    for idx, a in enumerate(A):
        C[idx] -= a * quotient
    C.pop(0)

print(*reversed(ans))
